olm/catalogd/v1alpha1/clustercatalog_types.go

```diff
@@ -0,0 +1,219 @@
+	//
+	// If unspecified, the default value is "Enabled"
+	//
+	// +kubebuilder:validation:Enum="Disabled";"Enabled"
+	// +kubebuilder:default="Enabled"
+	// +optional
+	Availability string `json:"availability,omitempty"`
+}
+
+// ClusterCatalogStatus defines the observed state of ClusterCatalog
+type ClusterCatalogStatus struct {
+	// conditions is a representation of the current state for this ClusterCatalog.
+	// The status is represented by a set of "conditions".
+	//
+	// Each condition is generally structured in the following format:
+	//   - Type: a string representation of the condition type. More or less the condition "name".
+	//   - Status: a string representation of the state of the condition. Can be one of ["True", "False", "Unknown"].
+	//   - Reason: a string representation of the reason for the current state of the condition. Typically useful for building automation around particular Type+Reason combinations.
+	//   - Message: a human-readable message that further elaborates on the state of the condition.
+	//
+	// The current set of condition types are:
+	//   - "Serving", which represents whether or not the contents of the catalog are being served via the HTTP(S) web server.
+	//   - "Progressing", which represents whether or not the ClusterCatalog is progressing towards a new state.
+	//
+	// The current set of reasons are:
+	//   - "Succeeded", this reason is set on the "Progressing" condition when progressing to a new state is successful.
+	//   - "Blocked", this reason is set on the "Progressing" condition when the ClusterCatalog controller has encountered an error that requires manual intervention for recovery.
+	//   - "Retrying", this reason is set on the "Progressing" condition when the ClusterCatalog controller has encountered an error that might be resolvable on subsequent reconciliation attempts.
+	//   - "Available", this reason is set on the "Serving" condition when the contents of the ClusterCatalog are being served via an endpoint on the HTTP(S) web server.
+	//   - "Unavailable", this reason is set on the "Serving" condition when there is not an endpoint on the HTTP(S) web server that is serving the contents of the ClusterCatalog.
+	//
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
+	// resolvedSource contains information about the resolved source based on the source type.
+	//
+	// Below is an example of a resolved source for an image source:
+	// resolvedSource:
+	//
+	//  image:
+	//    lastSuccessfulPollAttempt: "2024-09-10T12:22:13Z"
+	//    ref: quay.io/operatorhubio/catalog@sha256:c7392b4be033da629f9d665fec30f6901de51ce3adebeff0af579f311ee5cf1b
+	//  type: Image
+	// +optional
+	ResolvedSource *ResolvedCatalogSource `json:"resolvedSource,omitempty"`
+	// urls contains the URLs that can be used to access the catalog.
+	// +optional
+	URLs *CatalogURLs `json:"urls,omitempty"`
```

Comment:
This particular field is still a work in progress. The general gist of this field is to capture any URLs we think may be useful to users when crafting their requests to the HTTP server that is serving this `ClusterCatalog`'s content.

For more information on this field, see https://docs.google.com/document/d/1XvQPwk-Ws1rVcBpNsAaDoYKlmlws-hiQ19Ex9D4zAok/edit?usp=sharing

Reply:
Relevant types changes here merged in [operator-framework/catalogd#429](https://github.com/operator-framework/catalogd/pull/429) and released in v0.34.0 of catalogd.

Reply:
This section is now updated to match the most recent changes to the upstream types.